src/Stack/Types/Package.hs

```diff
@@ -92,29 +92,103 @@ newtype ExeName = ExeName { unExeName :: Text }
+          -- Likely derived from cabal see the field just before.
+          ,packageAllDeps :: !(Set PackageName)
+          -- ^ Original dependencies (not sieved).
+          -- The keyset of package deps.
+          ,packageGhcOptions :: ![Text]
+          -- ^ Ghc options used on package.
+          -- Comes from 'PackageConfig' 'packageConfigGhcOptions'.
+          -- Likely derived from cabal in
+          --
+          --    - buildTypeRaw function (see main comment)
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:options options function>
+          --
+          -- see 'generateBuildInfoOpts', and then all the way from loadPackage.
+          ,packageCabalConfigOpts :: ![Text]
+          -- ^ Additional options passed to ./Setup.hs configure
+          -- Absent in cabal ? Comes from 'PackageConfig' 'packageConfigCabalConfigOpts'.
+          ,packageFlags :: !(Map FlagName Bool)
+          -- ^ Flags used on package.
+          -- Absent in cabal ?
+          ,packageDefaultFlags :: !(Map FlagName Bool)
+          -- ^ Defaults for unspecified flags.
+          -- Absent in cabal ?
+          ,packageLibraries :: !PackageLibraries
+          -- ^ Does the package have a buildable library stanza?
+          -- Derived from the library and foreignLibs in Cabal.
+          -- If no main library or no buildable, we consider there is none.
+          --
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:foreignLibs foreignLibs>
+          --
+          ,packageInternalLibraries :: !(Set Text)
+          -- ^ names of internal libraries
+          -- Derived from cabal in:
+          --
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:subLibraries subLibraries>
+          --
+          ,packageTests :: !(Map Text TestSuiteInterface)
+          -- ^ names and interfaces of test suites
+          -- Derived from cabal in:
+          --
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:testSuites testSuites>
+          --
+          ,packageBenchmarks :: !(Set Text)
+          -- ^ names of benchmarks
+          -- Derived from cabal through:
+          --
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:benchmarks benchmarks>
+          --
+          ,packageExes :: !(Set Text)
+          -- ^ names of executables
+          -- Derived from cabal through:
+          --
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:executables executables>
+          --
+          ,packageOpts :: !GetPackageOpts
+          -- ^ Args to pass to GHC.
+          -- Absent in cabal.
```

Comment:
I guess for `Cabal` it's just a different type of a flag, see `--ghc-option` in  https://downloads.haskell.org/ghc/8.10.3/docs/html/libraries/Cabal-3.2.1.0/Distribution-Simple-Flag.html#t:Flag